cmake/match.py

```diff
@@ -22,10 +22,17 @@
     input_lines = input.readlines()
     match_lines = match.readlines()
 
-if len(match_lines) != len(input_lines):
-    sys.exit(f"Line count doesn't match (is: {len(input_lines)}, expected: {len(match_lines)})")
+if len(match_lines) < len(input_lines):
+    sys.exit(f"Match length < input length (input: {len(input_lines)}, match: {len(match_lines)})")
 
+input_idx = 0
+optional_line = False
```

Comment:
shouldn't this setting be done within `for` loop?

// I know we don't have a case where we mix opt with non-opt lines, but just to be sure 😉

Reply:
Done.